x-pack/filebeat/input/httpjson/internal/v2/input.go

```diff
@@ -0,0 +1,225 @@
+	inputcursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/transport/tlscommon"
+	"github.com/elastic/beats/v7/libbeat/common/useragent"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/go-concert/ctxtool"
+	"github.com/elastic/go-concert/timed"
+)
+
+const (
+	inputName = "httpjsonv2"
+)
+
+var (
+	userAgent = useragent.UserAgent("Filebeat")
+
+	// for testing
+	timeNow = time.Now
+)
+
+type retryLogger struct {
+	log *logp.Logger
+}
+
+func newRetryLogger() *retryLogger {
+	return &retryLogger{
+		log: logp.NewLogger("httpjson.retryablehttp", zap.AddCallerSkip(1)),
+	}
+}
+
+func (log *retryLogger) Error(format string, args ...interface{}) {
+	log.log.Errorf(format, args...)
+}
+
+func (log *retryLogger) Info(format string, args ...interface{}) {
+	log.log.Infof(format, args...)
+}
+
+func (log *retryLogger) Debug(format string, args ...interface{}) {
+	log.log.Debugf(format, args...)
+}
+
+func (log *retryLogger) Warn(format string, args ...interface{}) {
+	log.log.Warnf(format, args...)
+}
+
+func newTLSConfig(config config) (*tlscommon.TLSConfig, error) {
+	if err := config.Validate(); err != nil {
+		return nil, err
+	}
+
+	tlsConfig, err := tlscommon.LoadTLSConfig(config.Request.SSL)
+	if err != nil {
+		return nil, err
+	}
+
+	return tlsConfig, nil
+}
+
+func test(url *url.URL) error {
+	port := func() string {
+		if url.Port() != "" {
+			return url.Port()
+		}
+		switch url.Scheme {
+		case "https":
+			return "443"
+		}
+		return "80"
+	}()
+
+	_, err := net.DialTimeout("tcp", net.JoinHostPort(url.Hostname(), port), time.Second)
+	if err != nil {
+		return fmt.Errorf("url %q is unreachable", url)
+	}
+
+	return nil
+}
+
+func run(
+	ctx v2.Context,
+	config config,
+	tlsConfig *tlscommon.TLSConfig,
+	publisher inputcursor.Publisher,
+	cursor *inputcursor.Cursor,
+) error {
+	log := ctx.Logger.With("url", config.Request.URL)
+
+	stdCtx := ctxtool.FromCanceller(ctx.Cancelation)
+
+	httpClient, err := newHTTPClient(stdCtx, config, tlsConfig, log)
+	if err != nil {
+		return err
+	}
+
+	requestFactory := newRequestFactory(config.Request, config.Auth, log)
+	pagination := newPagination(config, httpClient, log)
+	responseProcessor := newResponseProcessor(config.Response, pagination, log)
+	requester := newRequester(httpClient, requestFactory, responseProcessor, log)
+
+	trCtx := emptyTransformContext()
+	trCtx.cursor = newCursor(config.Cursor, log)
+	trCtx.cursor.load(cursor)
+
+	err = timed.Periodic(stdCtx, config.Interval, func() error {
+		log.Info("Process another repeated request.")
+
+		if err := requester.doRequest(stdCtx, trCtx, publisher); err != nil {
+			log.Errorf("Error while processing http request: %v", err)
+		}
+
+		if stdCtx.Err() != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	log.Infof("Context done: %v", err)
```

Comment:
I think the message could use little more detail to help users understand what it means or what the impact is.